wsd/Storage.cpp

```diff
@@ -367,12 +365,20 @@ void LocalStorage::uploadLocalFileToStorageAsync(const Authorization& /*auth*/,
         if (_isCopy && Poco::File(getRootFilePathUploading()).exists())
             FileUtil::copyFileTo(getRootFilePathUploading(), path);
 
+        const FileUtil::Stat stat(path);
+
         // update its fileinfo object. This is used later to check if someone else changed the
         // document while we are/were editing it
-        setLastModifiedTime(
-            Util::getIso8601FracformatTime(FileUtil::Stat(path).modifiedTimepoint()));
+        setLastModifiedTime(Util::getIso8601FracformatTime(stat.modifiedTimepoint()));
         LOG_TRC("New FileInfo modified time in storage " << getLastModifiedTime());
         res = UploadResult(UploadResult::Result::OK);
+
+        if (asyncUploadCallback)
```

Comment:
before this change, if there was an exception and if asyncUploadCallback is present, then we would call:

asyncUploadCallback(... res) with the res of: UploadResult::Result::FAILED, "Internal error" post-catch

but after we no longer do that, so we now have an initially FAILED res which is always overwritten by an OK before use, so no point in having res declared before the try-catch block if that change is intentional?

It feels like we just want to change this to return the size, so maybe leave the call to asyncUploadCallback where it was and just assign stat.size() to a local var that defaults to 0 and return that local on function exit instead?

Reply:
My concern was that even with an exception we are setting the state as `Complete`, when `FAILED` is more appropriate.

In the new code, with an exception it will be `FAILED` and only when we reached the end do we set it to `Complete`, right before returning the size.

Your suggestion of having a local for size (defaulted to 0) is fine, but regarding what we pass `asyncUploadCallback` in the event of an exception, it's orthogonal.